src/types/models/index.ts

```diff
@@ -166,6 +166,25 @@ export interface SearchMessageLead extends MessageLead {
   createdDate: string
 }
 
+export interface SearchCallLead {
+  id: number
+  listingId: number
+  listing: {
+    externalListingId: string
+    firstRegistrationDate: string
+    id: number
+    image: string
+    make: string
+    mileage: number
+    model: string
+    price: number
+    type: string
+  }
```

Comment:
is this exactly the same as for `SearchMessageLead`? Can it be extracted to a shared interface so it's easier to align changes between the two?

Reply:
Good point. There is two that are not same but at the same time, even do they are in the API docs in swagger we don't actually get that data:
```
    firstRegistrationDate: string
    id: number
```
I'll extract and make those two optional.

Reply:
Changed it. Extracted lead listing par to `MessageLeadListing` interface.